cirq-google/cirq_google/engine/stream_manager.py

```diff
@@ -325,16 +346,15 @@ def _is_retryable_error(e: google_exceptions.GoogleAPICallError) -> bool:
     return any(isinstance(e, exception_type) for exception_type in RETRYABLE_GOOGLE_API_EXCEPTIONS)
 
 
-# TODO(#5996) Add stop signal to the request iterator.
 async def _request_iterator(
     request_queue: asyncio.Queue,
 ) -> AsyncIterator[quantum.QuantumRunStreamRequest]:
     """The request iterator for Quantum Engine client RPC quantum_run_stream().
 
     Every call to this method generates a new iterator.
     """
-    while True:
-        yield await request_queue.get()
+    while (request := await request_queue.get()) != StreamManager._STOP_SIGNAL:
```

Comment:
I'd suggest using `None` as the stop signal instead, then can simplify this a bit:
```suggestion
    while request := await request_queue.get():
```
Would also be good to explicitly type the queue:
```python
    self._request_queue: asyncio.Queue[Optional[QuantumRunStreamRequest]] = ...
```